app/controllers/landing_page_controller.rb

```diff
@@ -183,12 +191,8 @@ def parse_int(int_str_or_nil)
     nil
   end
 
-  def community_id(request)
-    request.env[:current_marketplace]&.id
-  end
-
-  def community_default_locale(request)
-    request.env[:current_marketplace]&.default_locale
+  def current_community(request)
+    @current_community ||= request.env[:current_marketplace]
```

Comment:
Great point, I did this to make it work quickly :) I can dig in to see how much data is needed from `@current_community`.